x-pack/plugin/deprecation/qa/early-deprecation-rest/src/javaRestTest/java/org/elasticsearch/xpack/deprecation/EarlyDeprecationIndexingIT.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.deprecation;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.apache.http.HttpHost;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestClientBuilder;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.test.rest.ESRestTestCase;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static org.elasticsearch.common.logging.DeprecatedMessage.KEY_FIELD_NAME;
+import static org.elasticsearch.xpack.deprecation.DeprecationTestUtils.DATA_STREAM_NAME;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.hamcrest.Matchers.startsWith;
+
+/**
+ * Tests that deprecation message on startup creates a deprecation data stream
+ */
+public class EarlyDeprecationIndexingIT extends ESRestTestCase {
+
+    /**
+     * In EarlyDeprecationTestPlugin#onNodeStarted we simulate a very early deprecation that can happen before the template is loaded
+     * The indexing has to be delayed until templates are loaded.
+     * This test confirms by checking the settings that a data stream is created (not an index with defaults)
+     * and that an early deprecation warning is not lost
+     */
+    public void testEarlyDeprecationIsIndexedAfterTemplateIsLoaded() throws Exception {
+
+        assertBusy(() -> {
+            Response response = getIndexSettings();
+            ObjectMapper mapper = new ObjectMapper();
+
+            final JsonNode jsonNode = mapper.readTree(response.getEntity().getContent());
```

Comment:
I think we can remove the `JsonNode ` usage here as well? And use `XContentMapValues `?